vpr/src/place/analytical_placement/PartialPlacement.cpp

```diff
@@ -93,11 +101,128 @@ double PartialPlacement::get_HPWL() {
                 max_y = node_loc_y[node_id];
             if (node_loc_y[node_id] < min_y)
                 min_y = node_loc_y[node_id];
+            if (max_x < min_x || max_y < min_y) {
+                VTR_LOG("maxx:%f, minx:%f, maxy: %f, miny: %f, x: %f, y: %f\n", max_x, min_x, max_y, min_y, node_loc_x[node_id], node_loc_y[node_id]);
+            }
+            VTR_ASSERT(max_x >= min_x);
+            VTR_ASSERT(max_y >= min_y);
         }
-        VTR_ASSERT(max_x >= min_x);
-        VTR_ASSERT(max_y >= min_y);
         hpwl += max_x - min_x + max_y - min_y;
     }
     return hpwl;
 }
 
+bool PartialPlacement::is_valid_partial_placement(){
+    VTR_ASSERT(node_loc_x.size() == node_loc_y.size());
```

Comment:
Since this method is used for debugging, I recommend making this an if statement and printing the message and return false.

Since this method will likely be used in an assert statement, it is a bit strange to have an assert in it. Also we may want to do something in the case the placement is invalid.